File: app/src/main/java/ch/blitzrechnen/app/data/Pin.kt
package ch.blitzrechnen.app.data

import java.security.MessageDigest

/**
 * Wandelt eine PIN in einen SHA-256-Hash um (Hex).
 * So wird die PIN nie im Klartext gespeichert.
 */
fun pinHash(pin: String): String {
    val bytes = MessageDigest.getInstance("SHA-256").digest(pin.trim().toByteArray())
    return bytes.joinToString("") { "%02x".format(it) }
}

/** Prüft, ob die eingegebene PIN zum gespeicherten Hash passt. */
fun checkPin(pin: String, storedHash: String?): Boolean =
    storedHash != null && pinHash(pin) == storedHash

File: expo/src/pin.ts
import * as Crypto from 'expo-crypto';

/** SHA-256-Hash der PIN (Hex) – die PIN wird nie im Klartext gespeichert. */
export async function pinHash(pin: string): Promise<string> {
  return Crypto.digestStringAsync(Crypto.CryptoDigestAlgorithm.SHA256, pin.trim());
}

export async function checkPin(pin: string, storedHash: string | null): Promise<boolean> {
  if (!storedHash) return false;
  const h = await pinHash(pin);
  return h === storedHash;
}
